overlord/devicestate/remodel.go

```diff
@@ -238,6 +255,40 @@ func (rc *baseRemodelContext) SystemMode() string {
 	return rc.systemMode
 }
 
+func (rc *baseRemodelContext) setRecoverySystemLabel(label string) {
+	rc.recoverySystemLabel = label
+}
+
+// updateRunModeSystem updates the device context used during boot and makes a
+// record of the new seeded system.
+func (rc *baseRemodelContext) updateRunModeSystem() error {
+	if rc.model.Grade() == asserts.ModelGradeUnset {
+		// nothing special for non-UC20 systems
+		return nil
+	}
+	if rc.recoverySystemLabel == "" {
+		return fmt.Errorf("internal error: recovery system label is unset during remodel finish")
+	}
+	// for UC20 systems we need record the fact that a new model is used for
+	// booting and consider a new recovery system as as seeded
+	oldDeviceContext := rc.GroundContext()
+	newDeviceContext := &rc.groundDeviceContext
+	if err := boot.DeviceChange(oldDeviceContext, newDeviceContext); err != nil {
+		return fmt.Errorf("cannot switch device: %v", err)
```

Comment:
Is there anything interesting about oldDeviceContext and newDeviceContext we could include in the error message?

Reply:
we could put some information from the model assertion there, otoh this is emitted in case a remodel from some model to another one fails, so the context should be clear here and we would be repeating things that are known already